### ISSUE TITLE
IPV6 Support - AssertionError [ERR_ASSERTION]: ipv4-mapped ipv6 addresses are currently unsupported!

### ISSUE THREAD

**Issue by Kief5555** [homebridge/homebridge#3637]
### Analysis

[6/24/2024, 7:23:58 PM] Homebridge v1.8.3 (HAP v0.12.2) (Homebridge 6DBB) is running on port 51148.
[6/24/2024, 7:24:12 PM] AssertionError [ERR_ASSERTION]: ipv4-mapped ipv6 addresses are currently unsupported!
    at enlargeIPv6 (/usr/local/lib/node_modules/homebridge/node_modules/@homebridge/ciao/src/util/domain-formatter.ts:137:9)
    at getNetAddress (/usr/local/lib/node_modules/homebridge/node_modules/@homebridge/ciao/src/util/domain-formatter.ts:296:26)
    at Function.HAPConnection.getLocalNetworkInterface (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/util/eventedhttp.ts:879:28)
    at new HAPConnection (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/util/eventedhttp.ts:394:43)
    at EventedHTTPServer.onConnection (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/util/eventedhttp.ts:231:24)
    at Server.emit (node:events:518:28)
    at TCP.onconnection (node:net:2197:8)
[6/24/2024, 7:24:12 PM] Error: read ECONNRESET
    at TCP.onStreamRead (node:internal/stream_base_commons:217:20)

On start.

### Expected Behavior

For it to run

### Steps To Reproduce

1. On MacOS with IPv6 Install 
2. Run it

### Logs

```markdown
[6/24/2024, 7:23:58 PM] Homebridge v1.8.3 (HAP v0.12.2) (Homebridge 6DBB) is running on port 51148.
[6/24/2024, 7:24:12 PM] AssertionError [ERR_ASSERTION]: ipv4-mapped ipv6 addresses are currently unsupported!
    at enlargeIPv6 (/usr/local/lib/node_modules/homebridge/node_modules/@homebridge/ciao/src/util/domain-formatter.ts:137:9)
    at getNetAddress (/usr/local/lib/node_modules/homebridge/node_modules/@homebridge/ciao/src/util/domain-formatter.ts:296:26)
    at Function.HAPConnection.getLocalNetworkInterface (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/util/eventedhttp.ts:879:28)
    at new HAPConnection (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/util/eventedhttp.ts:394:43)
    at EventedHTTPServer.onConnection (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/util/eventedhttp.ts:231:24)
    at Server.emit (node:events:518:28)
    at TCP.onconnection (node:net:2197:8)
[6/24/2024, 7:24:12 PM] Error: read ECONNRESET
    at TCP.onStreamRead (node:internal/stream_base_commons:217:20)
[6/24/2024, 7:24:12 PM] Got SIGTERM, shutting down Homebridge...
[6/24/2024, 7:24:17 PM] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null
[6/24/2024, 7:24:22 PM] [HB Supervisor] Restarting Homebridge...
[6/24/2024, 7:24:22 PM] [HB Supervisor] Starting Homebridge with extra flags: -I
[6/24/2024, 7:24:22 PM] [HB Supervisor] Started Homebridge v1.8.3 with PID: 33437
[6/24/2024, 7:24:22 PM] Loaded config.json with 0 accessories and 1 platforms.
[6/24/2024, 7:24:22 PM] Loaded 0 cached accessories from cachedAccessories.
[6/24/2024, 7:24:22 PM] ---
[6/24/2024, 7:24:23 PM] Loaded plugin: homebridge-config-ui-x@4.56.3
[6/24/2024, 7:24:23 PM] Registering platform 'homebridge-config-ui-x.config'
[6/24/2024, 7:24:23 PM] ---
[6/24/2024, 7:24:23 PM] Loading 1 platforms...
Setup Payload:
```

### Configuration

```json
{
    "bridge": {
        "name": "Homebridge",
        "username": "redacted",
        "port": 51148,
        "pin": "redacted",
        "advertiser": "bonjour-hap"
    },
    "accessories": [],
    "platforms": [
        {
            "name": "Config",
            "port": 8581,
            "platform": "config"
        }
    ]
}
```

### Environment

- OS: MacOS 15.0 B2
- Software: Homebridge Latest
- Node: 20.12.2
- npm: 10.5.0

### Process Supervisor

hb-service

**Comment by donavanbecker** [comment]
this is due to https://github.com/homebridge/ciao/issues/41

**Comment by Kief5555** [comment]
Gotcha. Il close in favour of #41

**Comment by donavanbecker** [comment]
@Kief5555, have another person reporting this #3640

If you update to the alpha build it will resolve it for now. Just remember these are alpha builds...

**Comment by donavanbecker** [comment]
Hey @Kief5555, we have upgrading to Homebridge beta branch? latest Homebridge beta version should be `1.8.4-beta.13`

**Comment by Kief5555** [comment]
> Hey @Kief5555, we have upgrading to Homebridge beta branch? latest Homebridge beta version should be `1.8.4-beta.13`

Hey, are you able to provide commands to update via brew? I don't have time to manually compile.

**Comment by donavanbecker** [comment]
Update from the Homebridge UI

**Comment by Kief5555** [comment]
Got it. Updated and the issues appears to be resolved :)

**Comment by donavanbecker** [comment]
Good to hear. We will have this in a future release of Homebridge. Please stay on beta branch until then.

**Comment by Kief5555** [comment]
> Good to hear. We will have this in a future release of Homebridge. Please stay on beta branch until then.

Sounds good. Thanks for the hard work. 🙏🏻

**Comment by Kief5555** [comment]
Hey @donavanbecker, don't want to annoy you with another issue - but this may be related (in terms of networking), has nothing to do with ipv6. But I'm getting this error now:

`WARNING couldn't map socket coming from remote address undefined:undefined     at local address ::192.168.0.66 to an interface!`

**Comment by donavanbecker** [comment]
That should be fixed in Homebridge v2.0.0

**Comment by Kief5555** [comment]
> That should be fixed in Homebridge v2.0.0

Ah gotcha. Can't update, my plugins dont support it quite yet.
